### PR TITLE
Added Chatroom and the chat server as well

### PR DESCRIPTION
**Pull request by Sach24eth**
- Created a new test branch to avoid merge conflicts due to different file-folder structures
- chat server is within the project folder itself for the moment. Might have to add Sahil's firebase server-side stuff too
- run chat server first, "npm run devStart" and then test.
- Will keep improving the way the chatroom looks.
- Search functionality at the messages on the home screen is pending still post discussion with Tejas and Sahil yesterday. I will be working on that as well

### CODE REVIEW

File: src/Components/Messages/MessageChatCard/MessageChatCard.jsx [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
+  console.log("chat prop:", props.chat)
+  console.log("username:", props.userName)
+  //console.log("GenreData:", props.allGenres)
   const chatIcon =
     props.chat.title.split("")[0].toUpperCase() +
     props.chat.title.split("")[1].toUpperCase();
+    const handleNavigation = () => {
+      console.log("Window open!!")
+      //setRoom(true)
+      window.open(`http://localhost:3000/chatrooms?room=${props.chat.title}`)
```
Comment (troytomasch) [inline src/Components/Messages/MessageChatCard/MessageChatCard.jsx]: I'm not sure I like how this opens a new tab. Is it possible to open the chatroom without opening a new tab?

Comment (Sach24eth) [reply src/Components/Messages/MessageChatCard/MessageChatCard.jsx]: Yeah, I don't mind changing it. I'll just redirect to another route on click.

**Review by tejasraibagi26 — CHANGES_REQUESTED** [review]
Messages are now being sent without delays. The only issue i noticed is that if i am logged in using email and password, the chat room wont open, it'll only display chatroom name (image attached). Works as expected when logged in via google
Screen Shot 2022-05-06 at 11 47 39 AM
.

**Comment by Sach24eth** [comment]
> Messages are now being sent without delays. The only issue i noticed is that if i am logged in using email and password, the chat room wont open, it'll only display chatroom name (image attached). Works as expected when logged in via google Screen Shot 2022-05-06 at 11 47 39 AM .

As discussed, this must be an issue with "Display name" not being set. If needed, we can get on a call and fix this asap.

**Pull request merged**